### 1. Import all Packages

In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

### 2. Read the data

In [9]:
visa_df = pd.read_csv(r"C:\Users\D.JYOTSNA\OneDrive\Documents\Full Stack Data Science\2. DataFiles\Visadataset.csv")
visa_df.

case_id continent education_of_employee has_job_experience  \
0  EZYV01      Asia           High School                  N   
1  EZYV02      Asia              Master's                  Y   
2  EZYV03      Asia            Bachelor's                  N   
3  EZYV04      Asia            Bachelor's                  N   
4  EZYV05    Africa              Master's                  Y   

  requires_job_training  no_of_employees  yr_of_estab region_of_employment  \
0                     N            14513         2007                 West   
1                     N             2412         2002            Northeast   
2                     Y            44444         2008                 West   
3                     N               98         1897                 West   
4                     N             1082         2005                South   

   prevailing_wage unit_of_wage full_time_position case_status  
0         592.2029         Hour                  Y      Denied  
1       83425.6500         Year                  Y   Certified  
2      122996.8600         Year                  Y      Denied  
3       83434.0300         Year                  Y      Denied  
4      149907.3900         Year                  Y   Certified

### 3. Divide categorical and numerical columns

In [13]:
cat = visa_df.select_dtypes(include = 'object').columns
num = visa_df.select_dtypes(exclude = 'object').columns
num

Index(['no_of_employees', 'yr_of_estab', 'prevailing_wage'], dtype='object')

## ========= Remember To Apply in Numerical Data Analysis =================

- min

- 25p

- mean

- median

- 50p

- 75p

- max

In [19]:
wage_data = visa_df['prevailing_wage']
wage_data

0           592.2029
1         83425.6500
2        122996.8600
3         83434.0300
4        149907.3900
            ...     
25475     77092.5700
25476    279174.7900
25477    146298.8500
25478     86154.7700
25479     70876.9100
Name: prevailing_wage, Length: 25480, dtype: float64

In [21]:
wage_count = len(wage_data)
wage_min = np.min(wage_data)
wage_max = np.max(wage_data)
wage_mean = np.mean(wage_data)
wage_median = np.median(wage_data)

In [25]:
a = [wage_count,wage_min,wage_max,wage_mean,wage_median]
cols = ['Count','Min','Max','Mean','Median']
pd.DataFrame(a,cols, columns = ['Prevailing Wage'] )

Prevailing Wage
Count      25480.000000
Min            2.136700
Max       319210.270000
Mean       74455.814592
Median     70308.210000

# Percentile and Quartile

In [29]:
p_25 = np.percentile(wage_data,25)
q_25 = np.quantile(wage_data,0.25)  # Ranges -1 to 1

In [31]:
q_25,p_25

(34015.479999999996, 34015.479999999996)

In [33]:
wage_count = len(wage_data)
wage_min = np.min(wage_data)
wage_max = np.max(wage_data)
wage_mean = np.mean(wage_data)
wage_median = np.median(wage_data)
p_25 = np.percentile(wage_data,25)
p_50 = np.percentile(wage_count,50)
p_75 = np.percentile(wage_data,75)

In [41]:
quartile_50 = round(np.quantile(wage_data,0.5),2)
quartile_50

70308.21

### Observation :-

- 50p = 70308

- 50 percentage of data is less than 70308

- 50 percentage of 25480 is less than 70308 wage

- 12740 applicants are have wage less than 70308

In [52]:
num = visa_df.select_dtypes(exclude = 'object').columns
wage_data = visa_df['prevailing_wage']
p_50 = np.percentile(wage_data,50)
con = wage_data < p_50
visa_df[con].head()

case_id      continent education_of_employee has_job_experience  \
0   EZYV01           Asia           High School                  N   
6   EZYV07           Asia            Bachelor's                  N   
7   EZYV08  North America            Bachelor's                  Y   
9   EZYV10         Europe             Doctorate                  Y   
12  EZYV13           Asia            Bachelor's                  Y   

   requires_job_training  no_of_employees  yr_of_estab region_of_employment  \
0                      N            14513         2007                 West   
6                      N             4985         1994                South   
7                      N             3035         1924                 West   
9                      N             2251         1995                South   
12                     N           123876         1963            Northeast   

    prevailing_wage unit_of_wage full_time_position case_status  
0          592.2029         Hour                  Y      Denied  
6        53635.3900         Year                  Y   Certified  
7          418.2298         Hour                  Y      Denied  
9        67514.7600         Year                  Y   Certified  
12       28663.0500         Year                  Y   Certified

In [56]:
wage_count = len(wage_data)
wage_min = np.min(wage_data)
wage_max = np.max(wage_data)
wage_mean = np.mean(wage_data)
wage_median = np.median(wage_data)
p_25 = np.percentile(wage_data,25)
p_50 = np.percentile(wage_count,50)
p_75 = np.percentile(wage_data,75)

a = [wage_count,wage_min,wage_max,wage_mean,wage_median,p_25,p_50,p_75]
cols = ['Count','Min','Max','Mean','Medina','25p','50p','75p']
pd.DataFrame(a,cols,columns = ['Prevailing Wage'])

Prevailing Wage
Count      25480.000000
Min            2.136700
Max       319210.270000
Mean       74455.814592
Medina     70308.210000
25p        34015.480000
50p        25480.000000
75p       107735.512500

In [70]:
num = visa_df.select_dtypes(exclude = 'object').columns
a = []
for i in num:
    b = visa_df[i]
    wage_count = len(b)
    wage_min = round(np.min(b),2)
    wage_max = round(np.max(b),2)
    wage_mean = round(np.mean(b),2)
    wage_median = round(np.median(b),2)
    p_25 = round(np.percentile(b,25),2)
    p_50 = round(np.percentile(b,50),2)
    p_75 = round(np.percentile(b,75),2)
    data = [wage_count,wage_min,wage_max,wage_mean,wage_median,p_25,p_50,p_75]
    a.append(data)    

cols = ['Count','Min','Max','Mean','Median','25p','50p','75p']
pd.DataFrame(a,columns = cols, index = ['no_of_employees', 'yr_of_estab', 'prevailing_wage']).T

no_of_employees  yr_of_estab  prevailing_wage
Count          25480.00     25480.00         25480.00
Min              -26.00      1800.00             2.14
Max           602069.00      2016.00        319210.27
Mean            5667.04      1979.41         74455.81
Median          2109.00      1997.00         70308.21
25p             1022.00      1976.00         34015.48
50p             2109.00      1997.00         70308.21
75p             3504.00      2005.00        107735.51

# Describe

In [74]:
visa_df.describe()

no_of_employees   yr_of_estab  prevailing_wage
count     25480.000000  25480.000000     25480.000000
mean       5667.043210   1979.409929     74455.814592
std       22877.928848     42.366929     52815.942327
min         -26.000000   1800.000000         2.136700
25%        1022.000000   1976.000000     34015.480000
50%        2109.000000   1997.000000     70308.210000
75%        3504.000000   2005.000000    107735.512500
max      602069.000000   2016.000000    319210.270000